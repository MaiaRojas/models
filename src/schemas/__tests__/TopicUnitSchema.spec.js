const mongoose = require('mongoose/browser');
const { TopicSchema, TopicUnitSchema } = require('../')(mongoose);
const babelTopicJson = require('./fixtures/topics/babel');

describe('TopicUnitSchema', () => {
  it('should fail validation when fields missing', () => {
    const doc = new mongoose.Document({}, TopicUnitSchema);
    expect(doc.validateSync().errors).toMatchSnapshot();
  });

  it('should pass sync validation ...', () => {
    const topic = new mongoose.Document({}, TopicSchema);
    const doc = new mongoose.Document({
      topic: topic._id,
      slug: '01-intro',
      title: 'Intro to Foo',
      stats: {
        partCount: 4,
        exerciseCount: 0,
        duration: 35,
        durationString: '35min',
      },
    }, TopicUnitSchema);
    expect(doc.validateSync()).toBe(undefined);
  });

  it('should validate existing topic json', () => {
    const topic = new mongoose.Document({}, TopicSchema);
    const units = Object.keys(babelTopicJson.syllabus).map(slug => ({
      topic: topic._id,
      slug,
      ...babelTopicJson.syllabus[slug],
    }));
    return Promise.all(units.map((unitJson) => {
      const unit = new mongoose.Document(unitJson, TopicUnitSchema);
      return unit.validate();
    }));
  });
});
