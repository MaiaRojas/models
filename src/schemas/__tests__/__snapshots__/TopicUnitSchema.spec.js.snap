// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopicUnitSchema should fail validation when fields missing 1`] = `
Object {
  "slug": [ValidatorError: Path \`slug\` is required.],
  "stats.duration": [ValidatorError: Path \`stats.duration\` is required.],
  "stats.durationString": [ValidatorError: Path \`stats.durationString\` is required.],
  "stats.exerciseCount": [ValidatorError: Path \`stats.exerciseCount\` is required.],
  "stats.partCount": [ValidatorError: Path \`stats.partCount\` is required.],
  "title": [ValidatorError: Path \`title\` is required.],
  "topic": [ValidatorError: Path \`topic\` is required.],
}
`;
